src/Microsoft.DocAsCode.Metadata.ManagedReference.Roslyn/Visitors/VisitorHelper.cs

```diff
@@ -60,7 +60,18 @@ public static string GetId(ISymbol symbol)
                 return typeof(object).FullName;
             }
 
-            return GetDocumentationCommentId(symbol)?.Substring(2);
+            var id = GetDocumentationCommentId(symbol)?.Substring(2);
+
+            if ((id is null) && (symbol is IFunctionPointerTypeSymbol functionPointerTypeSymbol))
+            {
+                // Roslyn doesn't currently support doc comments for function pointer type symbols
+                // This returns just the stringified symbol to ensure the source and target parts
+                // match for reference item merging.
+
+                return functionPointerTypeSymbol.ToString();
```

Comment:
As per the comment, Roslyn doesn't support doc comments for function pointers today. This looks to be more generally tracked by https://github.com/dotnet/roslyn/issues/46674.

This at least gives a type name that should be mostly unique and include all the relevant portions.